### PR TITLE
feat(utils): add timers to logging

### PR DESCRIPTION
**Pull request by ericssontadashi**
migrating from web3 logging

### CODE REVIEW

**Review by vasco-santos — COMMENTED** [review]
The set of changes look good to me, but would be great to know more about the motivation of this PR, so that I can better support in case other solutions exist.

As a side note, I would really prefer if we simply drop this file and used the common code in workers/utils https://github.com/web3-storage/workers/blob/master/packages/utils/src/logging.js

From what I see, it already has what is in here?

File: packages/api/src/utils/logs.js [hunk 1]
```diff
@@ -268,9 +298,25 @@ export class Logging {
     if (this.opts?.debug) {
       console.log(`[${this._date()}] `, `${name}: ${duration} ms`)
     }
-    return timeObj
```
Comment (alanshaw) [inline packages/api/src/utils/logs.js:271]: No return statement anymore?

File: packages/api/src/utils/logs.js [hunk 2]
```diff
@@ -237,28 +266,29 @@ export class Logging {
...
     this._timesOrder.push(name)
   }
 
   /**
    * @param {string} name
+   * @param {boolean} [shouldLog]
+   * @param {any} [context]
    */
-  timeEnd(name) {
+  timeEnd(name, shouldLog, context) {
```
Comment (alanshaw) [inline packages/api/src/utils/logs.js:284]: This should match the [`timeEnd`](https://developer.mozilla.org/en-US/docs/Web/API/console/timeEnd) method.

File: packages/api/src/utils/logs.js [hunk 3]
```diff
@@ -268,9 +298,25 @@ export class Logging {
     if (this.opts?.debug) {
       console.log(`[${this._date()}] `, `${name}: ${duration} ms`)
     }
-    return timeObj
+
+    if (shouldLog) {
```
Comment (alanshaw) [inline packages/api/src/utils/logs.js:302]: Note that timing execution is not effective in production and I believe that is why this method does not send logs to logtail.

https://developers.cloudflare.com/workers/learning/security-model/#step-1-disallow-timers-and-multi-threading

**Review by alanshaw — CHANGES_REQUESTED** [review]

**Comment by hugomrdias** [comment]
yes @ericssontadashi exactly what @vasco-santos said drop this file and use https://github.com/web3-storage/workers/blob/master/packages/utils/readme.md#logging directly if you have problems or questions doing that ping me

**Pull request closed without merging**